File: lab2task6/lab2task6/lab2task6/SerializeJson.cs
using System;
using System.Collections.Generic;
using System.IO;
using Newtonsoft.Json;

namespace lab2task6
{
    public class SerializeJson
    {
        public static void run(DeserializeJson deserializer, string destinationDir)
        {
            var newData = new List<dynamic>();
            foreach (var department in deserializer.Data)
            {
                newData.Add(new
                {
                    Kod_TERYT = department["Kod_TERYT"],
                    Województwo = department["Województwo"],
                    Powiat = department["Powiat"],
                    typ_JST = department["typ_JST"],
                    nazwa_urzędu_JST = department["nazwa_urzędu_JST"],
                    miejscowość = department["miejscowość"],
                    telefon_z_numerem_kierunkowym = $"{(int)department["telefon kierunkowy"]}{department["telefon"]}",
                });
            }
            var json = JsonConvert.SerializeObject(newData);
            File.WriteAllText(destinationDir, json);
        }
    }
}

File: lab2/lab2task2_5/serialize_json.py
import json


class SerializeJson:
    @staticmethod
    def run(deserializeddata, filelocation):
        print("let's serialize something")
        lst = []

        for dep in deserializeddata.data:
            lst.append({
                "Kod_TERYT": dep['Kod_TERYT'],
                "Województwo": dep['Województwo'],
                "Powiat": dep['Powiat'],
                "typ_JST": dep['typ_JST'],
                "nazwa_urzędu_JST":  dep['nazwa_urzędu_JST'],
                "miejscowość":  dep['miejscowość'],
                "telefon_z_numerem_kierunkowym":  str(dep["telefon kierunkowy"]) + dep['typ_JST']
            })

        jsontemp = {"departaments": lst}
        with open(filelocation, 'w', encoding='utf-8') as f:
            json.dump(jsontemp, f, ensure_ascii=False)